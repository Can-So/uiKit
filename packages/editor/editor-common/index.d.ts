export * from './utils';
export * from './types';
export * from './contextIdentifier';
export { default as ProviderFactory, WithProviders, Providers, } from './providerFactory';
export * from './styles';
export * from './ui';
