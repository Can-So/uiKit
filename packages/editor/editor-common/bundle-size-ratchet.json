[
  {
    "id": "src.main",
    "name": "main",
    "stats": {
      "size": 64416,
      "gzipSize": 16104
    }
  },
  {
    "id": "node_modules.main",
    "name": "node_modules [main]",
    "stats": {
      "size": 212338,
      "gzipSize": 62562
    }
  },
  {
    "id": "node_modules.async",
    "name": "node_modules [async]",
    "stats": {
      "size": 16063,
      "gzipSize": 6383
    }
  },
  {
    "id": "atlaskit.core.main",
    "name": "main",
    "stats": {
      "size": 102495,
      "gzipSize": 26415
    }
  },
  {
    "id": "atlaskit.editor.main",
    "name": "main",
    "stats": {
      "size": 44915,
      "gzipSize": 10570
    }
  },
  {
    "id": "atlaskit.elements.main",
    "name": "main",
    "stats": {
      "size": 19557,
      "gzipSize": 5632
    }
  },
  {
    "id": "atlaskit.elements.async",
    "name": "async",
    "stats": {
      "size": 7738,
      "gzipSize": 2890
    }
  },
  {
    "id": "atlaskit.people-and-teams.main",
    "name": "main",
    "stats": {
      "size": 68853,
      "gzipSize": 12328
    }
  },
  {
    "id": "combined.main",
    "name": "main",
    "stats": {
      "size": 502268,
      "gzipSize": 130935
    }
  },
  {
    "id": "combined.async",
    "name": "async",
    "stats": {
      "size": 23737,
      "gzipSize": 9031
    }
  }
]