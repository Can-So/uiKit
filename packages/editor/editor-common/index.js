export * from './utils';
export { default as ProviderFactory, WithProviders, } from './providerFactory';
export * from './styles';
export * from './ui';
//# sourceMappingURL=index.js.map