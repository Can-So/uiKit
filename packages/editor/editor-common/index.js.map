{"version":3,"file":"index.js","sourceRoot":"","sources":["src/index.ts"],"names":[],"mappings":"AAAA,cAAc,SAAS,CAAC;AAKxB,OAAO,EACL,OAAO,IAAI,eAAe,EAC1B,aAAa,GAEd,MAAM,mBAAmB,CAAC;AAE3B,cAAc,UAAU,CAAC;AACzB,cAAc,MAAM,CAAC"}