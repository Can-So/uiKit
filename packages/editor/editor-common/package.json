{
  "name": "@atlaskit/editor-common",
  "version": "9.4.0",
  "description": "A package that contains common classes and components for editor and renderer",
  "license": "Apache-2.0",
  "main": "dist/es5/index.js",
  "atlaskit:src": "src/index.ts",
  "jsnext:main": "dist/es2015/index.js",
  "module": "dist/es2015/index.js",
  "types": "dist/es5/index.d.ts",
  "sideEffects": false,
  "scripts": {
    "build:json-schema": "mkdir -p dist && cp -r json-schema dist",
    "generate:json-schema": "../../../node_modules/.bin/tsc -p build/json-schema && node ./dist/json-schema-cli/index.js ./src/schema/nodes/doc.ts > ./json-schema/v1/full.json"
  },
  "files": [
    "dist"
  ],
  "author": "Atlassian Pty Ltd",
  "maintainers": [
    "Oscar W",
    "Stanislav S"
  ],
  "repository": "https://bitbucket.org/atlassian/atlaskit-mk-2",
  "dependencies": {
    "@atlaskit/emoji": "^34.2.0",
    "@atlaskit/icon": "^11.3.0",
    "@atlaskit/media-card": "^27.1.4",
    "@atlaskit/mention": "^11.1.4",
    "@atlaskit/profilecard": "^3.7.1",
    "@atlaskit/util-shared-styles": "^2.10.3",
    "@types/prosemirror-model": "^1.0.0",
    "@types/prosemirror-view": "^1.0.0",
    "ajv": "^4.11.3",
    "classnames": "^2.2.5",
    "css-color-names": "0.0.4",
    "date-fns": "^1.29.0",
    "linkify-it": "^2.0.3",
    "prosemirror-model": "^1.2.0",
    "prosemirror-view": "^1.3.0",
    "raf-schd": "^2.1.0",
    "styled-components": "1.4.6 - 3",
    "typescript": "~2.6.2"
  },
  "peerDependencies": {
    "@atlaskit/media-core": "^18.0.3",
    "react": "^16.3.0",
    "react-dom": "^16.3.0"
  },
  "devDependencies": {
    "@atlaskit/editor-json-transformer": "^3.0.6",
    "@atlaskit/editor-test-helpers": "^3.1.8",
    "@atlaskit/media-core": "^18.0.3",
    "@atlaskit/webdriver-runner": "^0.0.3",
    "@types/chai": "^4.0.4",
    "chai": "^4.1.2",
    "json-schema-diff-validator": "^0.0.5",
    "sinon": "^2.2.0"
  }
}
