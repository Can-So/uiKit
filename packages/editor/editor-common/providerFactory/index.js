export { WithProviders } from './withProviders';
var ProviderFactory = /** @class */ (function () {
    function ProviderFactory() {
        this.providers = new Map();
        this.subscribers = new Map();
    }
    ProviderFactory.create = function (providers) {
        var providerFactory = new ProviderFactory();
        Object.keys(providers).forEach(function (name) {
            providerFactory.setProvider(name, providers[name]);
        });
        return providerFactory;
    };
    ProviderFactory.prototype.destroy = function () {
        this.providers.clear();
        this.subscribers.clear();
    };
    ProviderFactory.prototype.isEmpty = function () {
        return !this.providers.size && !this.subscribers.size;
    };
    ProviderFactory.prototype.setProvider = function (name, provider) {
        // Do not trigger notifyUpdate if provider is the same.
        if (this.providers.get(name) === provider) {
            return;
        }
        if (provider) {
            this.providers.set(name, provider);
        }
        else {
            this.providers.delete(name);
        }
        this.notifyUpdated(name, provider);
    };
    ProviderFactory.prototype.removeProvider = function (name) {
        this.providers.delete(name);
        this.notifyUpdated(name);
    };
    ProviderFactory.prototype.subscribe = function (name, handler) {
        var handlers = this.subscribers.get(name) || [];
        handlers.push(handler);
        this.subscribers.set(name, handlers);
        var provider = this.providers.get(name);
        if (provider) {
            handler(name, provider);
        }
    };
    ProviderFactory.prototype.unsubscribe = function (name, handler) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        var index = handlers.indexOf(handler);
        if (index !== -1) {
            handlers.splice(index, 1);
        }
        if (handlers.length === 0) {
            this.subscribers.delete(name);
        }
        else {
            this.subscribers.set(name, handlers);
        }
    };
    ProviderFactory.prototype.unsubscribeAll = function (name) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        this.subscribers.delete(name);
    };
    ProviderFactory.prototype.hasProvider = function (name) {
        return this.providers.has(name);
    };
    ProviderFactory.prototype.notifyUpdated = function (name, provider) {
        var handlers = this.subscribers.get(name);
        if (!handlers) {
            return;
        }
        handlers.forEach(function (handler) {
            handler(name, provider);
        });
    };
    return ProviderFactory;
}());
export default ProviderFactory;
//# sourceMappingURL=index.js.map