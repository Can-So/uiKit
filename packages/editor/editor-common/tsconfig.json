{
  "extends": "../../../tsconfig.editor.json",
  "include": [
    "src/**/*.ts",
    "src/**/*.tsx",
    "./docs/**/*.ts",
    "./docs/**/*.tsx",
    "test-helpers/**/*.ts",
    "test-helpers/**/*.tsx"
  ],
  "exclude": [
    "node_modules",
    "src/json-schema"
  ]
}
