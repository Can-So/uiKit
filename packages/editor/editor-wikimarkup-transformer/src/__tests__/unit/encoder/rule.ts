import { defaultSchema } from '@atlaskit/editor-common';
import WikiMarkupTransformer from '../../../index';

import { doc, hr, p } from '@atlaskit/editor-test-helpers';

describe('ADF => WikiMarkup - Panel', () => {
  const transformer = new WikiMarkupTransformer();

  test('should convert rule node', () => {
    const node = doc(
      p('This is a ruler'),
      hr(),
      p('I am in between a ruler'),
      hr(),
      p('I am under a ruler'),
    )(defaultSchema);
    expect(transformer.encode(node)).toMatchSnapshot();
  });
});
