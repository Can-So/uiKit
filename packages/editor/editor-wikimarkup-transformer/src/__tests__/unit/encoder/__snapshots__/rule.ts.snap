// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADF => WikiMarkup - Panel should convert rule node 1`] = `
"This is a ruler

----

I am in between a ruler

----

I am under a ruler"
`;
